components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -260,6 +268,10 @@ public ServiceProvider createApplicationWithTemplate(ServiceProvider serviceProv
                     LogConstants.ApplicationManagement.CREATE_APPLICATION_ACTION)
                     .data(buildSPData(serviceProvider));
             triggerAuditLogEvent(auditLogBuilder, true);
+
+            if (isTrustedAppConsentRequired() && isTrustedAppConsentGranted(serviceProvider)) {
```

Comment:
Since we dont control whether we allow to create the trusted app without consent from here we can do this independent from `isTrustedAppConsentGranted` we will simply log the value as either consented or not.

Reply:
I added this check so that this audit log will not get printed for applications not using this feature. Or else, since the default value is false even if a basic app is created, this log will get published.